proc/proc.go

```diff
@@ -0,0 +1,249 @@
+// proc offers control over arbitrary processes.
+// proc depends heavily on os/exec.
+// If/when proc needs to do resource control & isolation
+// its probable that os/exec will need to be replaced with
+// either the os package itself or syscall.
+package proc
+
+// Fingers crossed that its easier than rewriting os.ForkExec
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+const outputReadRefreshInterval = time.Millisecond * 10
+
+// Proc is a BitBox process.
+// Stderr and Stdout are dumped to temp files on disk.
+type Proc struct {
+	// TODO: there is 99% chance we need a synchronization primitive at this level
+	stdoutFileName string
+	stderrFileName string
+	cmd            *exec.Cmd
+}
+
+// NewProc constructs and begins process.
+// Stdout & Stderr of the new process are pointed at temp files.
+// The tempfileNames are acessable through the coresponding members.
+func NewProc(cmdName string, args ...string) (Proc, error) {
+
+	var err error
+	var cmdPath string
+	if cmdPath, err = exec.LookPath(cmdName); err != nil {
+		return Proc{}, err
+	}
+	cmd := exec.Command(cmdPath, args...)
+
+	var stdout *os.File
+	if stdout, err = ioutil.TempFile("", ""); err != nil {
+		return Proc{}, err
+	}
+	cmd.Stdout = stdout
+	stdoutName := stdout.Name()
+
+	var stderr *os.File
+	if stderr, err = ioutil.TempFile("", ""); err != nil {
+		return Proc{}, err
+	}
+	cmd.Stderr = stderr
+	stderrName := stderr.Name()
```

Comment:
it might be a better idea to combine stdout and stderr in the same file, just to get nice unified timeline of the output of the process

Reply:
Both of these solutions would work in my opinion. but I would say do not bother messing up with timestamps, reading and sorting files.... one file would work great for this particular scenario

Reply:
I agree. Done.